### DISCUSSION TITLE
How do you recommend to use bibliographic citations?

### DISCUSSION THREAD

**Discussion by Ypot** [toshism/org-super-links#74] [General]
It seems to me that it would be nice to have integrated the "cites" with the LINKS.

I am using org-ref time ago, but now it seems necessary some kind of integration.
For example, here there are two Zettels. One has an internal link and the other one a citation:

```
*** La mayor gloria para la conciencia es
:PROPERTIES:
:ID:       20200818T190646.436481
:END:
LINKS:
- [[id:Vieira.1994.700ExperimentosConscienciologia.C.38]]
:END
el autocontrol.

*** Parece como si para ejercer autocontrol y evitar caer en tentaciones, habría una
barra de energía que se va agotando. Y una vez agotada el comportamiento es guiado por los instintos. 
cite:Dewall.2013.ScientificSecretsSelfcontrol
```

**Comment by toshism** [comment]
I'm afraid I am not much help here as I have never used org-ref. Do you have a particular suggestion on how they could/should be integrated?

**Comment by Ypot** [comment]
> I'm afraid I am not much help here as I have never used org-ref. Do you have a particular suggestion on how they could/should be integrated?

No idea. Maybe org-roam integration as in org-roam-bibtex could guide us.

But I would suggest you to try org-ref if you want to add references to your books' title in your zettels. Even a basic use, makes you feel that's the way.

Shall we invoke @novoid?

**Comment by novoid** [comment]
Sorry, I was not able to set up org-ref myself and there was not much pressure to do so so far.

Further, I do not understand the question yet.

**Comment by Ypot** [comment]
Let's say, in your notes sometimes you link other notes. But other times you want to "link" the physical book you have on your hands.
I connect my note to that book by adding a clickable "cite" which, when you pass your cursor over, or click on it, offers you a complete information about that book, taken from a bibtex entry.
When exporting to PDF that cite and that referenced book are shown and compliant with citation formats too.

**How do you "link" books that are out of your system?**

Here org-roam-bibtex:
https://www.youtube.com/watch?v=Wy9WvF5gWYg

Here a bibtex entry as an example of the info in the bibliography file which can be fulfilled, with a few keystrokes, using a software like Zotero:
```
@book{Burns.1980.FeelingGood,
  title = {Feeling Good},
  author = {Burns, David D},
  year = {2012},
  publisher = {HarperCollins},
  abstract = {The good news is that anxiety, guilt, pessimism, procrastination, low self-esteem, and other?black holes? of depression can be cured without drugs. In Feeling Good, eminent psychiatrist David D. Burns, M.D. outlines the remarkable, scientifically proven techniques that will immediately lift your spirits and help you develop a positive outlook on life. Now, in this updated edition, Dr Burns adds an all-new Consumer?s Guide To Antidepressant Drugs, as well as a new introduction to help answer your questions about the many options available for treating depression. Recognise what causes your mood swings. Nip negative feelings in the bud. Deal with guilt. Handle hostility and criticism. Overcome addiction to love and approval. Build self-esteem. Feel good everyday.},
  isbn = {978-0-06-213649-7},
  language = {English}
}
```

Maybe these "cite:" links can be compatible with the links in the :DRAWER: or maybe org-ref is unnecessary and you are using a self-containing method to cite books, compatible with org links.

**Comment by Ypot** [comment]
> 
> 
> Sorry, I was not able to set up org-ref myself and there was not much pressure to do so so far.
> 
> Further, I do not understand the question yet.

I think that maybe the references of your notes should be all of them in the :LINKS: drawer. Because I feel that the potential in postprocessing the :LINKS: is giant, showing the relations between notes, filtering them, etc. But this doesn't work fine with the abovementioned citations, which are out of the :DRAWER: and wont be postprocessed with the other links.

**Comment by novoid** [comment]
> How do you "link" books that are out of your system?

Ah, now I understand.

In my case, any book does have its Org heading containing all the meta-data about a book: purchase, summary, borrowed to somebody, and so forth. So this is just a simple link as any other link to a heading.

**Comment by Ypot** [comment]
> 
> 
> > How do you "link" books that are out of your system?
> 
> Ah, now I understand.
> 
> In my case, any book does have its Org heading containing all the meta-data about a book: purchase, summary, borrowed to somebody, and so forth. So this is just a simple link as any other link to a heading.

How do you fulfill all that data? Manually? Could you please share with us an example?

**Comment by novoid** [comment]
> How do you fulfill all that data? Manually? Could you please share with us an example?

Sure: when I wrote my thesis 10+ years ago, I used: https://karl-voit.at/2015/12/26/reference-management-with-orgmode/

Afterwards, it was just books I created manually, neglecting bibtex files more and more. However, I plan to write a PIM book within the next years. So I'll give org-ref another try some day. Whatever system I'm using, there will always be a heading for any reference (paper, book).

**Comment by yantar92** [comment]
> How do you fulfill all that data? Manually? Could you please share with us an example?

FYI. I have WIP project aiming to retrieve metadata (including bibtex entry) when bookmarking webpages (including papers or books). The plan is to integrate it with org-ref without a need to treat .bib files as the main bibliography store.

https://github.com/yantar92/org-capture-ref/blob/master/org-capture-ref.el

**Comment by Ypot** [comment]
> 
> 
> > How do you fulfill all that data? Manually? Could you please share with us an example?
> 
> FYI. I have WIP project aiming to retrieve metadata (including bibtex entry) when bookmarking webpages (including papers or books). The plan is to integrate it with org-ref without a need to treat .bib files as the main bibliography store.
> 
> https://github.com/yantar92/org-capture-ref/blob/master/org-capture-ref.el

So we could add ebooks bibliography data to .org files, and use that info for make cites using org-ref while we can link to the entries by assigning an :ID: to each of them?

**Comment by yantar92** [comment]
> So we could add ebooks bibliography data to .org files, and use that info for make cites using org-ref while we can link to the entries by assigning an :ID: to each of them?

Not yet (WIP). The idea is to have `:BIBTEX:` drawer associated with any captured link (including links to books or articles). The contents of the drawer will be tangled into the .bib file (used by org-ref). Hence, usual org-ref mechanisms should work. 

I am currently thinking about the following capabilities:

1. Auto-create bibtex entry for *any* captured link (that's what I am working now);
2. The :ID: of the captured heading can be made equal to its cite key (hence, knowing only bibtex entry is sufficient to find the corresponding heading);
3. The bibtex entry is tangled into org-ref .bib file;
4. Org-ref note function will be redefined to open the right headline (it is easy as soon as bibtex key is also the :ID: of the headline)

At the end I hope to get equivalent of org-ref paper database, but org-centric (instead of .bib-centric database in native org-ref).

**Comment by novoid** [comment]
> At the end I hope to get equivalent of org-ref paper database, but org-centric (instead of .bib-centric database in native org-ref).

Sounds just like the thing I was missing when reading about org-ref!

**Comment by yantar92** [comment]
> I'm afraid I am not much help here as I have never used org-ref. Do you have a particular suggestion on how they could/should be integrated?

org-ref inserts a link to entry (or multiple entries) via `bibtex-completion-format-citation-functions`. org-super-links can probably hook into that variable changing the default `org-ref-bibtex-completion-format-org` to org-supper-links' version. 

The backlinks should be created in the bibtex entry note heading. The heading can be found/created according to `org-ref-notes-function`.

**Comment by Ypot** [comment]
Thank you all. For now I will stick to pure links to the book reference in an org file, and I will use "cite" just when I need to export a cite into latex.
So far, this would be the .org information about the book:
```
** Quintanilla Cobián \& Laura (2019). /Fundamentos de investigación en psicología/ :UNED:
   :PROPERTIES:
   :ID:       QuintanillaCobian.2019.FundamentosInvestigacionPsicologia
   :END:
   :LINKS:
   - [[id:20200828T210627.892589][Note about Quintanilla's book]]
   :END:
cite:QuintanillaCobian.2019.FundamentosInvestigacionPsicologia
```
The .bib registration:
```
@book{QuintanillaCobian.2019.FundamentosInvestigacionPsicologia,
  title = {Fundamentos de investigación en psicología},
  author = {{Quintanilla Cobián} and {Laura}},
  year = {2019},
  publisher = {UNED},
  address = {Madrid},
  isbn = {978-84-362-7556-8},
  language = {Spanish},
  note = {OCLC: 1127385164}
}
```

And a note related to the book with a link to the .org reference:
```
* Note about Quintanilla's book
:PROPERTIES:
:ID:       20200828T210627.892589
:END:
:LINKS:
- [[id:QuintanillaCobian.2019.FundamentosInvestigacionPsicologia]]
:END:
Here I would add a cite just if needed cite:QuintanillaCobian.2019.FundamentosInvestigacionPsicologia
```

**Comment by novoid** [comment]
@Ypot For the sake of completeness, you might want to check out https://github.com/novoid/extract_pdf_annotations_to_orgmode where I describe the method I was using almost a decade ago to extract highlighted phrases (via Android tablet and PDF annotating app) directly into Org. The tool might not work any more (not testes for years) but the principle does.

**Comment by Ypot** [comment]
> 
> 
> @Ypot For the sake of completeness, you might want to check out https://github.com/novoid/extract_pdf_annotations_to_orgmode where I describe the method I was using almost a decade ago to extract highlighted phrases (via Android tablet and PDF annotating app) directly into Org. The tool might not work any more (not testes for years) but the principle does.

Thanks. Repligo was my favorite PDF reader, since it allowed to see images while using "reflow mode".
Now I am using Pocket Book or others (no one as good as Repligo could be nowadays), I export from the reader into a file (usually html) and "copy-paste" the highlights and notes into my org entry, related to the book. Maybe I could use Webtools instead :m

I don't understand lisp, so I can hardly follow your code, but it seems that in your `references.org` you link to the .PDF book and the .BIB entry. I see in `references.org` no `:ID:` property added so I think you are using `${HOME}/bin/vkextract_annotations_to_orgmode_snippet.sh Voit2011` to link the note to the `references.org` entry. No backlink in `references.org` either, so I infer you are relying on your elisp skills to be able to filter all the notes related to a paper. Sorry I can't follow you better.

**Comment by novoid** [comment]
> I don't understand lisp

There is (almost no) no elisp involved. The repo consists of a shell script which I included in a babel snippet (which is part of my "add a reference" yasnippet). When executed, the script takes the PDF paper (following my file name convention `<author><year>.pdf` and my paper directory. It extracts the contained highlighting and the result is shown in the result of the babel code.
My `reference.org` contains links to the bib file, the pdf file and the annotated pdf file.
